### ISSUE TITLE
Full-Text indexes not working when an analyzer is specified.

### ISSUE THREAD

**Issue by mrksph** [neo4j/neo4j#12662]
I discovered that when creating a Full-Text index if a config parameter is passed to specify which analyzer to use then the Full-Text index won't work, although it will be shown as ONLINE when querying :schema.

- Neo4j version: 4.2.0
- Operating system: Windows 10 (Neo4j Desktop)
- API/Driver: Cypher (Neo4j Browser)

### Steps to reproduce
I've already reported this bug in Neo4j Community's forum here: https://community.neo4j.com/t/full-text-is-created-and-online-but-is-not-working/32064

Assume a simple data model like (Client) with some properties: 'internalId', a unique 'code', a 'name', etc... And several nodes populating the database.

1. Create UNIQUENESS Constraint on code property for Client nodes.

> CREATE INDEX ClientName FOR (c:Client) ON (c.name);

2. Create Full-Text Index on code property for Client specifying an analyzer, as shown in Neo4j Official Docs (https://neo4j.com/docs/cypher-manual/4.2/administration/indexes-for-full-text-search/#administration-indexes-fulltext-search-create-and-configure):

>  CALL db.index.fulltext.createNodeIndex("ClientCodeFT", ["Client"], ["code"], { analyzer: "standard-folding" })

3. Query the nodes using the Full-Text index:

> CALL db.index.fulltext.queryNodes('ClientCodeFT', '\*SOMECODE*') YIELD node
> RETURN node LIMIT 25;

The query returns (no changes, no records)

4. Drop the previously created Full-Text index:

> CALL db.index.fulltext.drop("ClientCodeFT");

5. Create it again, but this time without the config parameter:

>  CALL db.index.fulltext.createNodeIndex("ClientCodeFT", ["Client"], ["code"])

6. Query the nodes using the Full-Text index

> CALL db.index.fulltext.queryNodes('ClientCodeFT', '\*SOMECODE*') YIELD node
> RETURN node LIMIT 25;

The query will return nodes containing 'SOMECODE' in 'code' (if there's any).

### Expected behavior
It should return all nodes with label 'Client' that contains 'SOMECODE' in code property.

### Actual behavior
The query returns (no changes, no records)

**Comment by burqen** [comment]
Hi @mrksph! Thank you for reporting.
Do you have a few example of what the code property could look like, to help reproduce the issue?

**Comment by mrksph** [comment]
Hi @burqen 
Sure

Here's the output of for "Client" nodes "code" property 
> CALL apoc.meta.nodeTypeProperties()

```
:`Client`:`Entity`" | ["Client", "Entity"] | "code" | ["String"] | true | 1893 | 1893
```

And here's a fragment of the cypher script I use to create the schema and load the data.



Note: This FTS index worked in Neo4j v3.5. In my database I have nodes with label "Client". All "Client" nodes have one more label of 3 possibles. 

```
:begin
CREATE INDEX ON :Client(name);
:commit
:begin
CALL db.index.fulltext.createNodeIndex("ClientCodeFT",["Client"],["code"], { analyzer: "standard-folding" });
:commit
```

**Comment by mrksph** [comment]
I can reproduce this bug easily from the Neo4j browser.
1. Create the Uniqueness Constraint on the property
2. Create the FullText Index specifying analyzer on the same property
3. Query the nodes using the FTS index.
4. Note no results are returned.
5. Drop the FTS index
6. Create it again, this time without the analyzer
7. Run the same query using the FTS index
8. Watch returned nodes.

**Comment by burqen** [comment]
@mrksph I've been able to reproduce the issue with your steps and working on a solution 👍

**Comment by mrksph** [comment]
@burqen Thanks for the feedback, I started thinking it was a problem on my side! I'll keep an eye on it.

**Comment by burqen** [comment]
@mrksph The fix has been merged and back-ported to 4.0 and forward and will be part of next patch on all of those releases. Thank you again for reporting!
I'm closing this issue here, please feel free to re-open if needed.

**Comment by mrksph** [comment]
@burqen thanks! I can't see any commits regarding this issue, just out of curiosity why is that? I'd like to see the changes if possible.

Thanks!

**Comment by burqen** [comment]
@mrksph They are merged to the private repo and will be mirrored to the public one soon. Commit message is "Fulltext StandardFoldingAnalyzer correctly normalize search string"

**Comment by mrksph** [comment]
@burqen Hey I just saw the commit. We were planning to use 4.2.1 in production before this bug. In which version is this bug fixed?

**Comment by burqen** [comment]
@mrksph The bug is fixed in upcoming 4.2.4 patch.
